base/download.jl

```diff
@@ -11,7 +11,17 @@ if Sys.iswindows()
         client = "New-Object System.Net.Webclient"
         # in the following we escape ' with '' (see https://ss64.com/ps/syntax-esc.html)
         downloadfile = "($client).DownloadFile('$(replace(url, "'" => "''"))', '$(replace(filename, "'" => "''"))')"
-        run(`$ps -NoProfile -Command "$tls12; $downloadfile"`)
+        # PowerShell v3 or later is required for Tls12
+        try
+            run(`$ps -Version 3 -NoProfile -Command "$tls12; $downloadfile"`)
+        catch e
+            if e isa PipelineError{Process} && e.pipe.exitcode % Int32 == -393216
+                # appears to be "wrong version" exit code, based on
+                # https://docs.microsoft.com/en-us/azure/cloud-services/cloud-services-startup-tasks-common
+                error("Downloading files requires Windows Management Framework 3.0 or later.")
+            end
+            rethrow(e)
+        end
```

Comment:
This should probably be a separate commit.

Reply:
Same PR okay?

Reply:
Sure.

Reply:
In that case, should we export `pipeline_error` (or some sort of "throw error based on exit code" function?)